src/nvim/buffer_defs.h

```diff
@@ -649,7 +649,7 @@ struct file_buffer {
   bool b_p_swf;                 /* 'swapfile' */
   long b_p_smc;                 /* 'synmaxcol' */
   char_u      *b_p_syn;         /* 'syntax' */
-  long b_p_ts;                  /* 'tabstop' */
+  int b_p_ts;                   // 'tabstop'
```

Comment:
it creates a great deal of warnings if I keep it to long. See #3803 for my change of numerical options to int. There is no problem in this PR for now but indeed this is not ideal and the changes in window.c will hopefully change that if it's deemed good enough.